translator/internaldata/resource_to_oc.go

```diff
@@ -106,17 +107,41 @@ func internalResourceToOC(resource pdata.Resource) (*occommon.Node, *ocresource.
 	return &ocNode, &ocResource
 }
 
-func attributeValueToString(attr pdata.AttributeValue) string {
+func attributeValueToString(attr pdata.AttributeValue, jsonLike bool) string {
 	switch attr.Type() {
+	case pdata.AttributeValueNULL:
+		if jsonLike {
+			return "null"
+		}
+		return ""
 	case pdata.AttributeValueSTRING:
+		if jsonLike {
+			return fmt.Sprintf("%q", attr.StringVal())
+		}
 		return attr.StringVal()
+
 	case pdata.AttributeValueBOOL:
 		return strconv.FormatBool(attr.BoolVal())
+
 	case pdata.AttributeValueDOUBLE:
 		return strconv.FormatFloat(attr.DoubleVal(), 'f', -1, 64)
+
 	case pdata.AttributeValueINT:
 		return strconv.FormatInt(attr.IntVal(), 10)
+
+	case pdata.AttributeValueMAP:
+		// OpenCensus attributes cannot represent maps natively. Convert the
+		// map to a JSON-like string.
+		var kvs []string
+		m := attr.MapVal()
+		m.ForEach(func(k string, v pdata.AttributeValue) {
+			kvs = append(kvs, fmt.Sprintf("%q:%s", k, attributeValueToString(v, true)))
+		})
+		return "{" + strings.Join(kvs, ",") + "}"
```

Comment:
https://golang.org/pkg/strings/#Builder is more efficient

Reply:
Also is there any need to worry about escaping `,` in the key or value or are these "JSON-like strings" not meant to be strictly parsed?

Reply:
1. Changed to strings.Builder.

2. Escaping is handled correctly by %q formatter for the key and by recursive call to attributeValueToString for the values. Updated the tests to include backslash and double quote in the key and in the value.